cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfRobotComponent.cpp

```diff
@@ -55,3 +52,31 @@ void UUrdfRobotComponent::createChildComponents(UrdfLinkDesc* parent_link_desc,
         createChildComponents(child_link_desc, child_link_component);
     }
 }
+
+void UUrdfRobotComponent::applyKeyInputs()
+{
+    UInputComponent* InputComponent = GetOwner()->InputComponent;
+
+    for (auto& input_axis_binding : input_axis_bindings_) {
+        float axis_value = InputComponent->GetAxisValue(input_axis_binding.axis_name_);
+
+        switch (input_axis_binding.type_) {
+        case AxisBindingType::Set:
+            if (axis_value > 0) {
+                for (uint32 i = 0; i < input_axis_binding.component_names_.size(); i++) {
+                    UUrdfJointComponent* joint_component = joint_components_.at(input_axis_binding.component_names_[i]);
+                    joint_component->applyAction(input_axis_binding.values_[i]);
+                }
+            }
+            break;
+        case AxisBindingType::Add:
+            for (uint32 i = 0; i < input_axis_binding.component_names_.size(); i++) {
+                UUrdfJointComponent* joint_component = joint_components_.at(input_axis_binding.component_names_[i]);
+                joint_component->applyAction(joint_component->getAction() + axis_value * input_axis_binding.values_[i]);
```

Comment:
Check if the axis value is > 0 outside the switch statement. Additive keyboard actions should do nothing if the key isn't being pressed.

Reply:
fixed